packages/core/src/codecatalyst/devEnv.ts

```diff
@@ -11,6 +11,45 @@ import { Timeout } from '../shared/utilities/timeoutUtils'
 import { showMessageWithCancel } from '../shared/utilities/messages'
 import { isCloud9 } from '../shared/extensionUtilities'
 import { getLogger } from '../shared/logger'
+import { CodeCatalystAuthenticationProvider } from './auth'
+import { getThisDevEnv } from './model'
+
+let didStartDevEnvActivityHeartbeat: boolean = false
+/**
+ * Trys to start the Dev Env Activity Heartbeat to keep the Dev Env from timing out.
+ * If this successfully starts, subsequent calls to this function will do nothing.
+ *
+ * ---
+ * Why do we need this?
+ *
+ * We need this function since there are times the heartbeat mechanism cannot be started. Eg auth
+ * is not connected in a Dev Env. So we need to try again elsewhere.
```

Comment:
is there a way to express the same info in a more concise (yet still accurate and complete) wording?

Reply:
Reduced doc string